app/models/version.rb

```diff
@@ -1,14 +1,28 @@
 # frozen_string_literal: true
 
-# Represents a version of a digital object.
+# Represents a version of a digital object with associated metadata.
 # All workflow steps occur on to a particular version.
 class Version
-  def initialize(druid:, version:)
+  def initialize(druid:, version:, metadata: nil)
     @druid = druid
     @version_id = version
+    @metadata = metadata # this is a hash of metadata values to be stored in the VersionMetadata table
   end
 
-  attr_reader :druid, :version_id
+  attr_reader :druid, :version_id, :metadata
+
+  def update_metadata
+    # if no metadata passed in (nil), do nothing
+    return unless metadata
+
+    # if metadata passed in is an empty hash, delete the version metadata record
+    if metadata.empty?
+      VersionMetadata.find_by(druid:, version: version_id)&.destroy
```

Comment:
I'm fuzzy on whether we should delete these VersionMetadata records or if it's fine to let lines 22-23 update them with empty metadata values. Maybe it doesn't matter?

Reply:
Yeah, it's not a big deal either way, you'd end up with an empty hash in that row I think if you don't do this delete... just tending to my general cleanup tendencies